src/lib.rs

```diff
@@ -67,36 +71,55 @@ impl SimpleLogger {
     }
 
     /// Set the 'default' log level.
+    ///
+    /// You can override the default level for specific modules and their sub-modules using [`with_module_level`]
+    ///
+    /// [`with_module_level`]: #method.with_module_level
     #[must_use = "You must call init() to begin logging"]
     pub fn with_level(mut self, level: LevelFilter) -> SimpleLogger {
         self.default_level = level;
         self
     }
 
-    /// Override the log level for specific module.
+    /// Override the log level for some specific modules.
+    ///
+    /// This sets the log level of a specific module and all its sub-modules.
+    /// When both the level for a parent module as well as a child module are set,
+    /// the more specific value is taken. If the log level for the same module is
+    /// specified twice, the resulting log level is implementation defined.
     ///
     /// # Examples
     ///
-    /// Change log level for specific crate:
+    /// Silence an overly verbose crate:
     ///
     /// ```no_run
     /// use simple_logger::SimpleLogger;
     /// use log::LevelFilter;
     ///
-    /// SimpleLogger::new().with_module_level("something", LevelFilter::Warn).init();
+    /// SimpleLogger::new().with_module_level("chatty_dependency", LevelFilter::Warn).init().unwrap();
     /// ```
     ///
-    /// Disable logging for specific crate:
+    /// Disable logging for all dependencies:
     ///
     /// ```no_run
     /// use simple_logger::SimpleLogger;
     /// use log::LevelFilter;
     ///
-    /// SimpleLogger::new().with_module_level("something", LevelFilter::Off).init();
+    /// SimpleLogger::new()
+    ///     .with_level(LevelFilter::Off)
+    ///     .with_module_level("my_crate", LevelFilter::Info)
+    ///     .init()
+    ///     .unwrap();
     /// ```
     #[must_use = "You must call init() to begin logging"]
     pub fn with_module_level(mut self, target: &str, level: LevelFilter) -> SimpleLogger {
-        self.module_levels.insert(target.to_string(), level);
+        self.module_levels.push((target.to_string(), level));
+
+        /* Normally this is only called in `init` to avoid redundancy, but we can't initialize the logger in tests */
```

Comment:
> but we can't initialize the logger in tests

this may be a dumb question, but why can't we?

Reply:
Because there are multiple tests and the logger can only be initialized once. Also the `init` method consumes the logger. The alternative would be to split out the parts of `init` we need for the tests into their own method, but then I saw it's only one line and took the path of shortest resistance.

Reply:
Got it, thanks for your explanation!